common/metrics/src/lib.rs

```diff
@@ -90,8 +104,44 @@ fn get_metrics_file<P: AsRef<Path>>(dir_path: &P, file_name: &str) -> File {
         .expect("Open metrics file failed")
 }
 
+pub fn gather_metrics() -> Vec<prometheus::proto::MetricFamily> {
+    let metric_families = prometheus::gather();
+    let mut kept_families: Vec<_> = Vec::new();
+    let mut total: usize = 0;
+    let mut kept_count: usize = 0;
+    for metric_family in metric_families {
+        let family_count = metric_family.get_metric().len();
+
+        if kept_count + family_count <= MAX_COUNTERS
+            || METRICS_TO_KEEP.contains(&metric_family.get_name())
+        {
+            kept_count += family_count;
+            kept_families.push(metric_family);
+        }
+        total += family_count;
+    }
+
+    if kept_count < total {
+        error!(
+            "Metrics: Too many counters >{}.  Kept: {}, Dropped: {}",
+            MAX_COUNTERS,
+            kept_count,
+            total - kept_count
+        );
+    }
+
+    // These metrics will be reported on the next pull
+    NUM_METRICS
+        .with_label_values(&["kept"])
+        .set(kept_count as i64);
+
+    NUM_METRICS.with_label_values(&["total"]).set(total as i64);
```

Comment:
do we want to return all families for now and swap it over at some point in the near future? or do we just wanna go for it?

Reply:
I just did some CT analysis, to see how we're doing.

We're at 800 currently with CT.  So, we can either limit at 1000 now, or let us build up our metrics from the different teams, and then switch.

Given we have 20 nodes we, only have room for... 10 more counters, without dimensions according to this estimation.  It seems to me we need to do some cutting.